README.md

```diff
@@ -73,15 +73,236 @@ Here are the results of our trained model.
+| Red TL # of Samples    | 1998                                       |  
+| Yellow TL # of Samples | 1998                                       |  
+| Green TL # of Samples  | 1998                                       |  
+| No TL # of Samples     | 5994                                       |  
+
+\* Notice that the number of samples of red, yellow, and green traffic lights is equal to each other 
+and the number of samples of images without traffic lights is a triple of that. 
+Such a balancing of the dataset is suggested in the
+[How to improve object detection](https://github.com/AlexeyAB/darknet#how-to-improve-object-detection)
+section of README file from [https://github.com/AlexeyAB/darknet](https://github.com/AlexeyAB/darknet) 
+repository by [AlexeyAB](https://github.com/AlexeyAB).  
+
+\** Notice that you cannot obtain the same level of "balancing" from the 
+[4Tzones Traffic Lights Dataset](https://yadi.sk/d/a1Kr8Wmg0zfa0A) and the use of 
+[`data_preparer.py`](utils/data_preparer.py). The process of creating the 
+[4Tzones Traffic Lights Augmented Dataset](https://yadi.sk/d/q2Yyy9PO2SrMKQ) involved several manual steps. 
+It can be roughly described as follows.
+Initially we had stored labeled images from different ROS bags in different folders. On each dataset, 
+we performed flipping, scaling, and balancing. For balancing the number of samples per class was set to about 2000.
+Then, the script randomly picked 1998 / 3 = 666 (5994 / 3 = 1998 for images without traffic lights) samples. 
+After all, we combined augmented images defived from all three ROS bags and got the 
+[4Tzones Traffic Lights Augmented Dataset](https://yadi.sk/d/q2Yyy9PO2SrMKQ).
+
+
+#### Data Preparer Script
+For data augmentation and conversion of labels to different formats we have created a 
+[`data_preparer.py`](utils/data_preparer.py) script. It is a quite sophisticated script that is capable of
+performing horizontal image flipping, image scaling, adjustment of brightness and contrast, 
+image resizing, dataset balancing, that is, making number of samples of red, yellow, and green lights equal to
+a specified value while creating triple of that for samples without traffic lights, random picking of the specified
+number of samples, and conversion of image annotations (bounding boxes) to several different label formats, such as
+a format used in the [Bosh Small Traffic Lights Dataset](https://hci.iwr.uni-heidelberg.de/node/6132),
+a format produced by [Yolo_mark](https://github.com/AlexeyAB/Yolo_mark) tool, 
+a format required by the [keras-yolo3](https://github.com/qqwweee/keras-yolo3),
+a format of labels of the 
+[Vatsal Srivastava's Traffic Lights Dataset](https://drive.google.com/file/d/0B-Eiyn-CUQtxdUZWMkFfQzdObUE/view?usp=sharing).
+The `data_preparer.py --help` command produces a help-message that 
+presents the comprehensive instructions on how to use
+the script. We strongly recommend to read it before feeding your data to the script.
```

Comment:
"....recommend reading it..."

Reply:
Addressed. Thank you very much for the thorough review!